nncf/experimental/openvino_native/engine.py

```diff
@@ -36,19 +36,21 @@ def __init__(self, model: ov.Model, target_device: TargetDevice = TargetDevice.C
         self.compiled_model = ie.compile_model(model, target_device.value)
         self.input_names = set(inp.get_friendly_name() for inp in model.get_parameters())
 
-    def infer(self, input_data: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
+    def infer(self, input_data: Union[np.ndarray, List[np.ndarray], Tuple[np.ndarray], Dict[str, np.ndarray]]) \
+              -> Dict[str, np.ndarray]:
         """
         Runs model on the provided input via OpenVINO Runtime.
         Returns the dictionary of model outputs by node names.
 
         :param input_data: inputs for the model.
         :return output_data: models outputs.
         """
-        model_inputs = {}
-        for name in self.input_names:
-            if name not in input_data:
-                raise RuntimeError(f'Missing a required input: {name} to run the model.')
-            model_inputs[name] = input_data[name]
-
-        model_outputs = self.compiled_model(model_inputs)
+        if len(self.input_names) != len(input_data):
+            raise RuntimeError(f'Model expects {len(self.input_names)} inputs, but {len(input_data)} are provided.')
+        if isinstance(input_data, dict):
+            for name in self.input_names:
+                if name not in input_data:
+                    raise RuntimeError(f'Missing a required input: {name} to run the model.')
```

Comment:
I don't think that we need to check all of this. I mean that if something is wrong with `input_data` we will get an error from `compiled_model`, won't we? @alexsu52 What do you think?

Reply:
In this case `compiled_model` raise error:
```
RuntimeError: Check '::getPort(port, name, {_impl->GetInputs(), _impl->GetOutputs()})' failed at src/inference/src/cpp/ie_infer_request.cpp:382:
Port for tensor name a was not found.
````

Reply:
I got it. It makes sense to have these checks. What do you think about moving the validation code to a separate static method? For example,
```python
@staticmethod
def _check_input_data_format(input_data, input_names):
```
My motivation is to separate logic from checks.

Reply:
I agree.